geniza/corpus/tests/test_import_data.py

```diff
@@ -362,6 +369,210 @@ def test_add_document_language():
+
+    # try fetching some of them with no cache
+    with caplog.at_level(logging.DEBUG):
+        assert get_user("Alan Elbaum") == alan  # firstname lastname
+        assert "Found user" in caplog.record_tuples[-1][2]
+        assert get_user("Olivia Remie Constable") == orc    # multi names
+        assert "Found user" in caplog.record_tuples[-1][2]
+        get_user("Olivia Remie Constable")  # second call, should be cached
+        assert "Using cached user" in caplog.record_tuples[-1][2]
+
+    # reset & add a name to the cache; should notify when cache is hit
+    import_data_cmd.user_lookup = {"AE": alan}
+    with caplog.at_level(logging.DEBUG):
+        assert get_user("Alan Elbaum") == alan  # cache miss
+        assert "Found user" in caplog.record_tuples[-1][2]
+        assert get_user("AE") == alan           # cache hit
+        assert "Using cached user" in caplog.record_tuples[-1][2]
+
+    # test fetching users from database using initials only; should cache
+    with caplog.at_level(logging.DEBUG):
+        assert get_user("ORC") == orc   # three initials
+        assert "Found user" in caplog.record_tuples[-1][2]
+        assert import_data_cmd.user_lookup["ORC"] == orc    # cached initials
+        assert get_user("MR") == marina  # two initials
+        assert "Found user" in caplog.record_tuples[-1][2]
+        assert import_data_cmd.user_lookup["MR"] == marina
+        get_user("MR")  # test cache using initials
+        assert "Using cached user" in caplog.record_tuples[-1][2]
+
+    # cache/database miss should warn and use the team user
+    with caplog.at_level(logging.WARNING):
+        assert get_user("asdf36") == import_data_cmd.team_user
+        assert "Couldn't find user" in caplog.record_tuples[-1][2]
+        assert get_user("") == import_data_cmd.team_user
+        assert "Couldn't find user" in caplog.record_tuples[-1][2]
+
+
+@pytest.mark.django_db
+@override_settings(DATA_IMPORT_URLS={})
+def test_get_edit_history(caplog):
+    import_data_cmd = import_data.Command()
+    import_data_cmd.setup()
+    get_edit_history = import_data_cmd.get_edit_history
+    User = get_user_model()
+
+    # empty dates returns empty list; nothing to do
+    history = get_edit_history("", "")
+    assert history == []
+
+    # users with no dates still does nothing; no way to log an event
+    history = get_edit_history("Sarah Nisenson", "")
+    assert history == []
+
+    # date with no user results in event assigned to whole team (TEAM_USER)
+    history = get_edit_history("", "5/4/2016")
+    assert history == [{
+        "type": "Created",
+        "user": import_data_cmd.team_user,
+        "date": datetime.date(2016, 5, 4),
+    }]
+
+    # simplest case: one user, one well-formed date -> creation event
+    history = get_edit_history("Sarah Nisenson", "3/16/2018")
+    assert history == [{
+        "type": "Created",
+        "user": User.objects.get(username="snisenson"),
```

Comment:
This is actually failing on the user comparison, not the date — different instances of the same django model must not be considered equal

It would be better to assert the comparisons separately so you can see where the error is when there's a mismatch. (And then you can just check username in the returned user instead of comparing to a different instance.)